Refactor sex related places

cuducos opened the pull request:
Thanks to @guilhermeprokisch we've got a great start in this topic.

This pulls request merges his contributions with some refactor @anaschwendler and I did the other day.

Still a WIP as we have to test it with the proper API key and as we have a new code review — thanks @irio — to take into account.

Irio commented:
What's missing here, @cuducos?

cuducos commented:
Some refactors/improvememts:

- [ ] Avoid loosing data when a request fail (concurrency issue on file writing, partially solved with `Queue` but I need to test it further)
- [ ] Enhance performance with more parallel requests (revert to `asyncio` commit but keep the queue for file writing)
- [ ] :nail_care: Standardize coroutines (discard `grequests` in favor of `asyncio`)

I feel like I still need to work on that since we haven't been able to download ghe full dataset yet. Unfortunately I'm more focused on Jarbas and on documenting Rosie and the toolbox right now.

cuducos commented:
Finally! This script is ready for code review 🎊 🍾 ✨ 🎉 

Thanks for the great work from @guilhermeprokisch that gave us a base for this data collection, thanks for numerous pair prigramming sessions with @jtemporal and @anaschwendler, we have a neat, fast and modern script collecting data.

I'm submitting it for code review and starting to collect data. As soon as I have the database (probably in six days due to API cap around 150k requests/day) I will:

* Create a sample notebook in this repo
* Upload the dataset do S3 and add it to the tool box

daneoshiga commented on src/fetch_sex_places.py:
```diff
@@ -0,0 +1,407 @@
+    async def get_closest(self):
+        """
+        Gets the closest place find in reads self.places, queries for its
+        details and returns a dictoinary with all the info on that place.
```
typo `dictoinary` and i would suggest using just `dict` when referencing the type.

cuducos commented:
Good news: script is ready. Bad news: dataset is basically worthless.

---

We don't have funds to gather **all** data: due to Google Places API pricing policy wither it would take ages to collect data using the free tier, or it would cost more than 300k USD. Thus the script now takes three filters: `term`, `value` and `city` (check `python src/fetch_sex_places.py --help` for details).

> Don't forget to `pip install -r requirements.txt` as we added `aiohttp` and `aiofiles` to the stack.

According to [a random sample that looked promising](https://twitter.com/cuducos/status/840882495868530688) the aforementioned dataset has data for a couple of cities — check 834ab3c for details.

The important thins however is that we have tons of _false positives_, so I added a notebook commenting on that — also check 834ab3c for details. 

---

Either way I believe these results — the data collection, the brief analysis and the notebook discussing the quality of data — can be merged into `master` as a documentation of the current status of the hypothesis raised in #39.

anaschwendler commented:
Took some time to look for this PR.
I already knew the algorithm for the solution, and for me it looks great.
For me it look real good to be merged, so if @jtemporal thinks the same, we can merge it.

jtemporal commented:
I agree with @anaschwendler I'm merging this one ;) great job everyone!

guilhermeprokisch commented:
To try to find more reliable results. I suggest filtering the companies name in the bills for some typical stooge names, like bars and hotels.

guilhermeprokisch commented:
But in the end the need a human confirmation that such place is stooge place. For this, we can call them or go there.
